packages/components/src/components/PasswordCreationField/PasswordCreationField.tsx

```diff
@@ -0,0 +1,365 @@
+    };
+
+    const rootClassName = clsx(
+      className,
+      formFieldStyles.formField,
+      styles.passwordField,
+    );
+
+    const propsContext: PropsContext = {
+      Button: {
+        tunnelId: "button",
+        size: "m",
+        variant: "plain",
+        color: "dark",
+        isDisabled: isDisabled,
+        className: dynamic((p) => clsx(p.className, styles.button)),
+      },
+      Label: {
+        optional: !isRequired,
+        className: formFieldStyles.label,
+        children: dynamic((localProps) => {
+          return (
+            <FieldLabel
+              locales={passwordCreationFieldLocales}
+              disabled={isDisabled}
+              onGeneratePasswordAction={onPasswordGenerateHandler}
+              ruleValidationResult={policyValidationResult?.ruleResults}
+              value={value}
+            >
+              {localProps.children}
+            </FieldLabel>
+          );
+        }),
+      },
+      FieldDescription: {
+        className: formFieldStyles.fieldDescription,
+      },
+      FieldError: {
+        className: formFieldStyles.customFieldError,
+        children: dynamic((p) => {
+          if (p.children) {
+            return p.children;
+          }
+
+          if (translatedStatusText) {
+            return translatedStatusText;
+          }
+        }),
+      },
+    };
+
+    return (
+      <ClearPropsContext>
+        <TunnelProvider>
+          <Aria.TextField
+            type={isPasswordRevealed ? "text" : "password"}
+            value={value}
+            onChange={onChange}
+            className={rootClassName}
+            isDisabled={isDisabled}
+            isInvalid={
+              isInvalid || (!!value && !statusTextFromValidationResult?.isValid)
+            }
+            isRequired={isRequired}
+            {...rest}
+          >
+            <Aria.Group
+              isDisabled={isDisabled}
+              className={clsx(styles.inputGroup)}
+            >
+              <Aria.Input
+                disabled={isDisabled}
+                onChange={onPasswordInputChangeHandler}
+                className={styles.input}
+                ref={ref}
+                value={value}
+              />
+              <Aria.Group className={styles.buttonContainer}>
+                <Action action={togglePasswordVisibilityHandler}>
+                  <Button
+                    data-component="toggleRevealPassword"
+                    isDisabled={isDisabled}
+                    className={styles.button}
+                    slot="button"
+                    size="m"
+                    variant="plain"
+                    color="dark"
+                    aria-label="Show password in plaintext"
+                  >
+                    {isPasswordRevealed ? <IconShow /> : <IconHide />}
+                  </Button>
+                </Action>
+                <TunnelExit id="button" />
+              </Aria.Group>
+              <div
```

Comment:
magst du das hier evtl noch ein bisschen aufteilen? zb die Complexity Anzeige in eine neue Datei auslagern? die Datei ist so arg riesig (aber ist vlt auch nur Geschmackssache^^)